packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.util.ts

```diff
@@ -383,11 +383,18 @@ export const generateClientErrorCodeForIceFailure = ({
     errorCode = MISSING_ROAP_ANSWER_CLIENT_CODE;
   }
 
-  if (signalingState === 'stable' && iceConnectionState === 'connected') {
+  if (
+    signalingState === 'stable' &&
+    (iceConnectionState === 'connected' || iceConnectionState === 'disconnected')
```

Comment:
Ah, great, so you've managed to do this fix without having to store any state. 
I think there is also a chance that the state may reach "failed" if our timeout is long enough, but probably we cannot assume that "failed" means that it was "connected" at some point, because I think both transitions are possible: connecting->failed and connecting->connected->disconnected->failed. 
Anyway this PR improves things and we can worry about "failed" state later if we find that it happens in the field often enough.

Reply:
Good.
Lets treat it as a short term solution, and we will be working on the long term